internal/org/repository/eventsourcing/eventstore.go

```diff
@@ -0,0 +1,187 @@
+
+	aggregates, err := orgCreatedAggregates(ctx, es.AggregateCreator(), org)
+
+	return org, aggregates, err
+}
+
+func (es *OrgEventstore) CreateOrg(ctx context.Context, orgModel *org_model.Org) (*org_model.Org, error) {
+	org, aggregates, err := es.PrepareCreateOrg(ctx, orgModel)
+	err = es_sdk.PushAggregates(ctx, es.PushAggregates, org.AppendEvents, aggregates...)
+	if err != nil {
+		return nil, err
+	}
+
+	return OrgToModel(org), nil
+}
+
+func (es *OrgEventstore) OrgByID(ctx context.Context, org *org_model.Org) (*org_model.Org, error) {
+	if org == nil {
+		return nil, errors.ThrowInvalidArgument(nil, "EVENT-gQTYP", "org not set")
+	}
+	query, err := OrgByIDQuery(org.AggregateID, org.Sequence)
+	if err != nil {
+		return nil, err
+	}
+
+	esOrg := OrgFromModel(org)
+	err = es_sdk.Filter(ctx, es.FilterEvents, esOrg.AppendEvents, query)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+	if esOrg.Sequence == 0 {
+		return nil, errors.ThrowNotFound(nil, "EVENT-kVLb2", "org not found")
+	}
+
+	return OrgToModel(esOrg), nil
+}
+
+func (es *OrgEventstore) DeactivateOrg(ctx context.Context, orgModel *org_model.Org) (*org_model.Org, error) {
+	if orgModel == nil {
+		return nil, errors.ThrowInvalidArgument(nil, "EVENT-oL9nT", "org not set")
+	}
+	org := OrgFromModel(orgModel)
+
+	aggregate := orgDeactivateAggregate(es.AggregateCreator(), org)
+	err := es_sdk.Push(ctx, es.PushAggregates, org.AppendEvents, aggregate)
+	if err != nil {
+		return nil, err
+	}
+
+	return OrgToModel(org), nil
+}
+
+func (es *OrgEventstore) ReactivateOrg(ctx context.Context, orgModel *org_model.Org) (*org_model.Org, error) {
+	if orgModel == nil {
+		return nil, errors.ThrowInvalidArgument(nil, "EVENT-9t73w", "org not set")
+	}
+	org := OrgFromModel(orgModel)
+
+	aggregate := orgReactivateAggregate(es.AggregateCreator(), org)
+	err := es_sdk.Push(ctx, es.PushAggregates, org.AppendEvents, aggregate)
+	if err != nil {
+		return nil, err
+
+	}
+	return OrgToModel(org), nil
+}
+
+func (es *OrgEventstore) OrgMemberByIDs(ctx context.Context, member *org_model.OrgMember) (*org_model.OrgMember, error) {
+	if member == nil || member.UserID == "" || member.AggregateID == "" {
+		return nil, errors.ThrowPreconditionFailed(nil, "EVENT-ld93d", "member not set")
+	}
+
+	org, err := es.OrgByID(ctx, &org_model.Org{ObjectRoot: member.ObjectRoot, Members: []*org_model.OrgMember{member}})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, currentMember := range org.Members {
+		if currentMember.UserID == member.UserID {
+			return currentMember, nil
+		}
+	}
+
+	return nil, errors.ThrowNotFound(nil, "EVENT-SXji6", "member not found")
+}
+
+func (es *OrgEventstore) PrepareAddOrgMember(ctx context.Context, member *org_model.OrgMember) (*OrgMember, *es_models.Aggregate, error) {
+	if member == nil || !member.IsValid() {
+		return nil, nil, errors.ThrowPreconditionFailed(nil, "EVENT-9dk45", "UserID and Roles are required")
+	}
+
+	repoMember := OrgMemberFromModel(member)
```

Comment:
we should read existing to make change on latest

Reply:
that's not possible because the org doesn't exist if I setup the org